src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -1727,12 +1791,444 @@ def get_no_wait(self, **kwargs) -> bool:
+        if (
+            self.mc and
+            self.mc.network_profile and
+            self.mc.network_profile.load_balancer_profile
+        ):
+            value_obtained_from_mc = (
+                self.mc.network_profile.load_balancer_profile.idle_timeout_in_minutes
+            )
+
+        # set default value
+        if value_obtained_from_mc is not None:
+            load_balancer_idle_timeout = value_obtained_from_mc
+        else:
+            load_balancer_idle_timeout = raw_value
+
+        # this parameter does not need dynamic completion
+        # this parameter does not need validation
+        return load_balancer_idle_timeout
+
+    # pylint: disable=unused-argument
+    def get_outbound_type(self, enable_validation: bool = False, **kwargs) -> Union[str, None]:
+        """Obtain the value of outbound_type.
+
+        Note: The parameters involved in the validation are not verified in their own getters.
+
+        When outbound_type is not assigned, dynamic completion will be triggerd. By default, the value is set to
+        CONST_OUTBOUND_TYPE_LOAD_BALANCER.
+
+        This function supports the option of load_balancer_profile, if provided, when verifying loadbalancer-related
+        parameters, the value in load_balancer_profile will be used for validation.
+        This function supports the option of enable_validation. When enabled, if the value of outbound_type is
+        userDefinedRouting (CONST_OUTBOUND_TYPE_USER_DEFINED_ROUTING), the following checks will be performed. If
+        load_balancer_sku is set to basic, an InvalidArgumentValueError will be raised. If vnet_subnet_id is not
+        assigned, a RequiredArgumentMissingError will be raised. If any of load_balancer_managed_outbound_ip_count,
+        load_balancer_outbound_ips or load_balancer_outbound_ip_prefixes is assigned, a MutuallyExclusiveArgumentError
+        will be raised.
+        This function supports the option of read_only. When enabled, it will skip dynamic completion and validation.
+
+        :return: string or None
+        """
+        # read the original value passed by the command
+        raw_value = self.raw_param.get(
+            "outbound_type"
+        )
+        # try to read the property value corresponding to the parameter from the `mc` object
+        value_obtained_from_mc = None
+        if (
+            self.mc and
+            self.mc.network_profile
+        ):
+            value_obtained_from_mc = (
+                self.mc.network_profile.outbound_type
+            )
+
+        # set default value
+        read_from_mc = False
+        if value_obtained_from_mc is not None:
+            outbound_type = value_obtained_from_mc
+            read_from_mc = True
+        else:
+            outbound_type = raw_value
+
+        # skip dynamic completion & validation if option read_only is specified
+        if kwargs.get("read_only"):
+            return outbound_type
+
+        dynamic_completion = False
```

Comment:
not necessary.

Reply:
Good catch, changed in commit [a3b6403](https://github.com/Azure/azure-cli/pull/19458/commits/a3b6403d7bcc51ae21c10143c1d960add2257f3b).